### ISSUE TITLE
Listing a mapping doesn't clear command line properly

### ISSUE THREAD

**Issue by zeertzjq** [vim/vim#9786]
### Steps to reproduce

1. Run `vim --clean`
2. Type `:nmap a b` and press enter
3. Type
```
:                    nmap a
```
and press enter
4. Command line becomes
```
n  a             b   nmap a
```

### Expected behaviour

Command line becomes
```
n  a             b
```

### Version of Vim

v8.2.4396

### Environment

Operating system: Arch Linux
Terminal: kitty
Value of $TERM: xterm-kitty
Shell: fish

**Comment by zeertzjq** [comment]
GitHub swallowed the spaces. There are twenty spaces after the first colon in step 3.

**Comment by lacygoill** [comment]
Yes, it's a confusing bug.  FWIW, I think it's a duplicate of #5623 and #5962.

There exists a relevant todo item:
https://github.com/vim/vim/blob/f5288c589500de0677444af4a428cfbccfccb8ce/runtime/doc/todo.txt#L424-L425

   > :map output does not clear the reset of the command line.
   > (#5623, also see #5962)

**Comment by zeertzjq** [comment]
Oh I didn't search carefully.

**Comment by lacygoill** [comment]
No worry, I didn't find it either when I posted a duplicate issue.

**Comment by lacygoill** [comment]
Fixed by patch [8.2.4401](https://github.com/vim/vim/releases/tag/v8.2.4401)
